File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> mp;
        int n = nums.size();
        for(int i=0;i<n;i++){
            mp[nums[i]]++;
        }

        int target = n/2;
        int res = -1;
        for(const auto& s:mp){
            if(s.second>target) res = s.first;
        }
        return res;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = {}
        n = len(nums)
        target = n // 2

        for num in nums:
            count[num] = count.get(num, 0) + 1

        for key, val in count.items():
            if val > target:
                return key